src/syncerd/bitcoin_syncer.rs

```diff
@@ -411,6 +413,107 @@ fn query_addr_history(
     Ok(addr_txs)
 }
 
+fn p2wpkh_script_code(script: &bitcoin::Script) -> Script {
+    bitcoin::blockdata::script::Builder::new()
+        .push_opcode(bitcoin::blockdata::opcodes::all::OP_DUP)
+        .push_opcode(bitcoin::blockdata::opcodes::all::OP_HASH160)
+        .push_slice(&script[2..])
+        .push_opcode(bitcoin::blockdata::opcodes::all::OP_EQUALVERIFY)
+        .push_opcode(bitcoin::blockdata::opcodes::all::OP_CHECKSIG)
+        .into_script()
+}
+
+fn sweep_address(
+    private_key: [u8; 32],
+    address: bitcoin::Address,
+    destination_address: bitcoin::Address,
+    electrum_server: String,
+) -> Result<Vec<Vec<u8>>, Error> {
+    match address.address_type() {
+        Some(bitcoin::AddressType::P2wpkh) => {}
+        Some(address_type) => {
+            return Err(Error::Farcaster(format!(
+                "Sweeping addresses only supports native segwit v1 addresses. Address has type: {}",
+                address_type
+            )));
+        }
+        None => return Err(Error::Farcaster(format!("Invalid to be swept address"))),
+    }
+
+    let sk = bitcoin::PrivateKey::from_slice(&private_key, bitcoin::Network::Testnet)?;
+    let pk = bitcoin::PublicKey::from_private_key(bitcoin::secp256k1::SECP256K1, &sk);
+
+    let client = Client::new(&electrum_server)?;
+    let unspent_txs = client.script_list_unspent(&address.script_pubkey())?;
+
+    if unspent_txs.len() == 0 {
+        return Ok(vec![]);
+    }
+
+    let in_amount = unspent_txs
+        .iter()
+        .fold(0, |acc, unspent_tx| acc + unspent_tx.value);
+    let inputs: Vec<bitcoin::TxIn> = unspent_txs
+        .iter()
+        .map(|unspent_output| bitcoin::TxIn {
+            previous_output: bitcoin::OutPoint {
+                txid: unspent_output.tx_hash,
+                vout: unspent_output.tx_pos as u32,
+            },
+            script_sig: bitcoin::Script::default(),
+            sequence: (1 << 31) as u32,
+            witness: bitcoin::Witness::new(),
+        })
+        .collect();
+
+    let mut unsigned_tx = bitcoin::Transaction {
+        version: 2,
+        lock_time: 0,
+        input: inputs,
+        output: vec![bitcoin::TxOut {
+            value: in_amount,
+            script_pubkey: destination_address.script_pubkey(),
+        }],
+    };
+    unsigned_tx.output[0].value = in_amount - 2 * unsigned_tx.vsize() as u64;
```

Comment:
no clue, 2 is the fee rate?

Reply:
Sorry, I should have left a comment there, since I wanted to discuss this. Yes, it is the fee rate. I don't know how I should handle it properly yet. Should we add a fee priority level to the SweepAddress task?

Reply:
In swapd i also do some ugly thing for the fees

personally i advocate for mempool position as the best way to prioritize txs

And yes, it does make sense to add priority level to sweepaddress